webroot/adm/oncall/generate-cstg-keypair.html

```diff
@@ -0,0 +1,636 @@
+                    clearFormsAndShowCollapsed([]);
+                    $("#searchSiteNoSiteFoundAlert").collapse('hide');
+                    $("#searchSiteErrorAlert").collapse('hide');
+                    $("#searchResultsTableBody").empty();
+
+                    const name = $("#participantName").val();
+
+                    doApiCallWithCallback("GET", "/api/site/list", (text) => {
+                        const searchList = JSON.parse(text);
+
+                        const similarList = searchList.filter((site) => site.name.toLowerCase().includes(name.toLowerCase()));
+                        if (similarList.length === 0) {
+                            $("#searchSiteNoSiteFoundAlert").collapse('show');
+                        } else {
+
+                            doApiCallWithCallback("GET", "/api/client_side_keypairs/list", (text) => {
+                                const keypairs = JSON.parse(text);
+
+                                similarList.forEach((site) => {
+                                    const siteId = site.id;
+                                    const siteKeyPairs = keypairs.filter((keypair) => keypair.site_id == siteId);
+                                    const siteJson = JSON.stringify(site, null, 2).replace('"id": ' + siteId, `<a href='#' siteIdData='${siteId}'>"id": ${siteId}</a>`);
+                                    const keypairJson = JSON.stringify(siteKeyPairs, null, 2).replace('"site_id": ' + siteId, `<a href='#' siteIdData='${siteId}'>"site_id": ${siteId}</a>`);
+                                    const rowToAdd = `<tr><td><pre>${siteJson}</pre></td><td><pre>${keypairJson}</pre></td></tr>`;
+
+                                    if ($("#searchResultsTableBody").length == 1) {
+                                        $("#searchResultsTableBody").append(rowToAdd);
+                                    } else {
+                                        $("#searchResultsTableBody tr:last").after(rowToAdd);
+                                    }
+                                })
+
+                                $('#searchResultsTableBody a').on("click", (e) => {
+                                const siteId =e.target.getAttribute('siteIdData');
+                                setSiteId(siteId);
+                                return false;
+                            })
+
+                            }, (err) => alertErrorCallback(err, "#searchOutputErrorDiv", "#searchSiteErrorAlert"));
+                        }
+                    }, (err) => alertErrorCallback(err, "#searchOutputErrorDiv", "#searchSiteErrorAlert"));
+                });
+
+                $("#createSite-workflow-submit").on("click", () => {
+                    let domainNames = convertNamesToArray($("#createSite-workflow-domainNames").val());
+                    if (domainNames.length === 0) {
+                        showCreateSiteError("Domain names cannot be empty");
```

Comment:
I don't think we should actually error here. A site needs app names OR domain names. I guess setting app names is beyond scope of this PR, so I think we can just allow `domainNames` to be empty here.

Reply:
Added AppNames